File: 69-sqrtx/69-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l,r=0,x
        while l<=r:
            mid=(l+r)//2
            if mid*mid <= x < (mid+1)*(mid+1):
                return mid
            elif x<mid*mid:
                r = mid-1
            else:
                l=mid+1
        

File: 69-sqrtx/69-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if(x==0||x==1)
            return x;
        int left=0;
        int right=x;
        while(left<=right)
        {
            int mid=(left+right)/2;
            if(mid==x/mid)
                return mid;
            if(mid>x/mid)
                right=mid-1;
            else
                left=mid+1;
        }
        return left-1;
    }
};